cmd/kubeadm/app/cmd/alpha/kubelet.go

```diff
@@ -109,16 +111,39 @@ func newCmdKubeletConfigDownload() *cobra.Command {
 			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
 			kubeadmutil.CheckErr(err)
 
-			err = kubeletphase.DownloadConfig(client, kubeletVersion, constants.KubeletRunDirectory)
+			criSocket, err := getCRISocket(criSocketStr)
```

Comment:
criSocketStr is always empty string here, right? Why not to pass it explicitly, i.e. ```criSocket, err := getCRISocket("")``` ?

Reply:
if you read the full code you will know how does kubelet assign value to criSocketStr.

Please read line 129:

> cmdutil.AddCRISocketFlag(cmd.Flags(), &criSocketStr)

Reply:
Actually, we don't need the extra var `criSocket` here. We can reuse `criSocketStr` and even have it renamed to `criSocket` only.